specviz/third_party/glue/viewer_single_spectrum.py

```diff
@@ -0,0 +1,170 @@
+
+from glue.utils.qt import load_ui
+
+from .utils import glue_data_to_spectrum1d, is_glue_data_1d_spectrum
+from ...widgets.main_window import MainWindow
+
+__all__ = ['SpecvizSingleDataViewer']
+
+
+class SpecvizSingleViewerState(ViewerState):
+
+    y_att = SelectionCallbackProperty(docstring='The attribute to use on the y-axis')
+
+    def __init__(self, *args, **kwargs):
+        super(SpecvizSingleViewerState, self).__init__(*args, **kwargs)
+        self._y_att_helper = ComponentIDComboHelper(self, 'y_att')
+        self.add_callback('layers', self._on_layers_change)
+
+    def _on_layers_change(self, value):
+        self._y_att_helper.set_multiple_data(self.layers_data)
+
+
+class SpecvizSingleLayerState(LayerState):
+    pass
+
+
+class SpecvizSingleLayerArtist(LayerArtist):
+
+    _layer_state_cls = SpecvizSingleLayerState
+
+    def __init__(self, specviz_window, *args, **kwargs):
+
+        super(SpecvizSingleLayerArtist, self).__init__(*args, **kwargs)
+
+        self.specviz_window = specviz_window
+        self.plot_widget = self.specviz_window.workspace.current_plot_window.plot_widget
+
+        self.state.add_callback('visible', self.update)
+        self.state.add_callback('zorder', self.update)
+        self._viewer_state.add_callback('y_att', self.update)
+        self.uuid = str(uuid.uuid4())
+
+    def _on_visible_change(self, value=None):
+        self.redraw()
+
+    def _on_zorder_change(self, value=None):
+        self.redraw()
+
+    def clear(self):
+
+        # FIXME: simplify the following by implementing DataListModel.remove_data
+        # and making it able to take e.g. data name.
+
+        data_model = self.specviz_window.workspace._model
+        for i in range(data_model.rowCount()):
+            item = data_model.item(i)
+            if item.data() == self.uuid:
+                data_model.removeRow(i)
+                return
+
+    def remove(self):
+        pass
+
+    def redraw(self):
+        pass
+
+    @property
+    def proxy_index(self):
+
+        # FIXME: this and plot_data_item definitely needs to be simplified!
+
+        data_model = self.specviz_window.workspace._model
+        for i in range(data_model.rowCount()):
+            item = data_model.item(i)
+            if item.data() == self.uuid:
+                break
+        else:
+            return None
+
+        source_index = self.plot_widget.proxy_model.sourceModel().indexFromItem(item)
+        proxy_index = self.plot_widget.proxy_model.mapFromSource(source_index)
+
+        return proxy_index
+
+    @property
+    def plot_data_item(self):
+        """
+        Get the PlotDataItem corresponding to this layer artist.
+        """
+
+        # FIXME: this definitely needs to be simplified!
+
+        proxy_index = self.proxy_index
+        if proxy_index is None:
+            return
+        else:
+            return self.specviz_window.workspace.proxy_model.item_from_index(proxy_index)
```

Comment:
@nmearl - here it would be nice to have a way to get a PlotDataItem given the dataset name (which here is a UUID)

Reply:
Would you prefer something like

```
self.specviz_window.workspace.plot_data_item_from_id(id)
```

or

```
self.specviz_window.workspace.proxy_model.item_from_id(id)
# And relatedly, for DataItems
self.specviz_window.workspace.model.item_from_id(id)
```